mods/beds/api.lua

```diff
@@ -97,6 +99,20 @@ function beds.register_bed(name, def)
 			type = "fixed",
 			fixed = {0, 0, 0, 0, 0, 0},
 		},
+		on_destruct = function(pos)
```

Comment:
Yes. Both nodes have a destruct function. If the top is destroyed, then it will trigger the bottom's `on_destruct` and will 'tell' it not to care about destroying the first node. Following the same system, if the bottom node is destroyed, it will trigger the top's `on_destruct` callback, which won't destroy recursively the first node.